cryptol-remote-api/src/CryptolServer.hs

```diff
@@ -1,43 +1,43 @@
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NoMonomorphismRestriction #-}
 module CryptolServer where
 
 import Control.Exception
 import Control.Lens
 import Control.Monad
 import Control.Monad.IO.Class
 import qualified Data.Aeson as JSON
+import Data.Foldable
+import qualified Data.Text as Text
+import qualified Data.Vector as Vector
 
 import Cryptol.Eval.Monad (EvalOpts(..), PPOpts(..))
-import Cryptol.ModuleSystem (ModuleCmd, ModuleEnv, checkExpr, evalExpr, loadModuleByPath, loadModuleByName)
-import Cryptol.ModuleSystem.Env (getLoadedModules, lmCanonicalPath, lmFingerprint, meLoadedModules, initialModuleEnv, meSolverConfig)
+import Cryptol.ModuleSystem
+  (ModuleCmd, ModuleEnv, ModuleError(..), ModuleWarning(..),
+   checkExpr, evalExpr, loadModuleByPath, loadModuleByName)
+import Cryptol.ModuleSystem.Env
+  (getLoadedModules, lmCanonicalPath, lmFingerprint, meLoadedModules,
+   initialModuleEnv, meSolverConfig)
 import Cryptol.ModuleSystem.Fingerprint
 import Cryptol.Parser.AST (ModName)
 import Cryptol.Utils.Logger (quietLogger)
 import Cryptol.Utils.PP (pretty)
 
 import Argo
 
-cantLoadMod :: JSON.Value -> JSONRPCException
-cantLoadMod mod =
-  makeJSONRPCException 3 "Can't load module" (Just mod)
-
-cryptolError :: JSON.Value -> JSONRPCException
-cryptolError mod =
-  makeJSONRPCException 4 "Cryptol error" (Just mod)
-
 runModuleCmd :: ModuleCmd a -> Method ServerState a
 runModuleCmd cmd =
     do s   <- getState
        out <- liftIO $ cmd (theEvalOpts, view moduleEnv s)
        case out of
          (Left x, warns) ->
-           raise (cryptolError (JSON.toJSON (pretty x, map pretty warns)))
+           raise (cryptolError x warns)
          (Right (x, newEnv), warns) ->
            do setState (set moduleEnv newEnv s)
               return x
 
-data LoadedModule = LoadedModule
+data LoadedModule = LoadedMoodule
```

Comment:
This new constructor name seems to be a typo...

Reply:
Mooooodule 🐮 

```suggestion
data LoadedModule = LoadedModule
```